apps/system/utils/kdbg_heapinfo.c

```diff
@@ -43,42 +48,101 @@ extern size_t regionx_size[CONFIG_MM_REGIONS];
 extern int regionx_heap_idx[CONFIG_MM_REGIONS];
 #endif
 
+static int heapinfo_fd;
+static int heapinfo_display_flag = HEAPINFO_DISPLAY_ALL;
+static int heapinfo_mode = HEAPINFO_SIMPLE;
+static int heapinfo_pid = HEAPINFO_PID_ALL;
 #if CONFIG_MM_NHEAPS > 1
 extern heapinfo_total_info_t total_info;
 #endif
 
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
-static void kdbg_heapinfo_init(FAR struct tcb_s *tcb, FAR void *arg)
+static void kdbg_heapinfo_set_option(char opt, int pid)
```

Comment:
1. I don't know what is adv. of separated function to do below.
  Even you find an option through getopt(), you use switch case here again.
2. You already give a value as pid argument when you call this function.
  But you don't use it when pid is all.
  Let's use `heapinfo_pid = pid;` for all of cases.

Reply:
for using options repeatedly with -k option.
getopt() reads 'k' only as a option, but we want to use 'a' configuration.